python/tests/test_highlevel.py

```diff
@@ -3867,6 +3867,21 @@ def test_branch_length_empty_tree(self):
         assert tree.branch_length(1) == 0
         assert tree.total_branch_length == 0
 
+    @pytest.mark.parametrize("root_threshold", [1, 2, 3])
+    def test_is_root(self, root_threshold):
+        # Make a tree with multiple roots with different numbers of samples under each
+        ts = tskit.Tree.generate_balanced(5).tree_sequence
+        ts = ts.decapitate(ts.max_root_time - 0.1)
+        tables = ts.dump_tables()
+        tables.nodes.add_row(flags=0)  # Isolated non-sample
+        tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE)  # Isolated sample
+        ts = tables.tree_sequence()
+        assert {ts.first().num_samples(u) for u in ts.first().roots} == {1, 2, 3}
+        tree = ts.first(root_threshold=root_threshold)
+        roots = set(tree.roots)
+        for u in range(ts.num_nodes):  # Will also test isolated nodes
+            assert tree.is_root(u) == (u in roots)
```

Comment:
This tests that the Python `is_root` definition is the same as the (C API's tree iterator + Python `roots`) definition for the `generate_balanced` case, but are there any other topologies that we should check that this holds?

It's not clear to me that visiting the siblings of the left root (as `Tree.roots` does) checks that the sibling meets the `root_threshold`? Maybe I'm missing something here.

Reply:
That's taken care of in the lower-level code @benjeffery - stuff only gets into the list of roots if it passes this test.

Reply:
As in, something can only become a child of the virtual root if it passes the "rootiness" test

Reply:
Ah, great. I assume this was taken care of, but thought it worth checking.